project/src/main/java/com/google/sps/data/RestaurantScorer.java

```diff
@@ -0,0 +1,34 @@
+package com.google.sps.data;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public final class RestaurantScorer {
+  /** Class should not be instantiated. */
+  private RestaurantScorer() {}
+
+  /** Returns the percent match for a restaurant to a set of user preferences. */
+  public static double score(Restaurant restaurant, JSONObject preferences) throws JSONException {
+    int preferredPriceLevel = preferences.getJSONObject("priceLevel").getInt("pref");
+    String preferredDiningExp = preferences.getJSONObject("diningExp").getString("pref");
+    int priceLevelWeight = preferences.getJSONObject("priceLevel").getInt("weight");
+    int diningExpWeight = preferences.getJSONObject("diningExp").getInt("weight");
+    int maxPoints = priceLevelWeight + diningExpWeight + 1;
+    double score = 0;
+    double restaurantRating = restaurant.getAvgRating();
+    if (restaurant.getPriceLevel() == preferredPriceLevel) {
+      score += priceLevelWeight;
+    }
+    if (restaurant.getPlaceTypes().contains(preferredDiningExp)) {
+      score += diningExpWeight;
+    }
+    if (restaurantRating >= 3) {
+      score += restaurantRating / 5;
```

Comment:
Lets pull 5 & 3 into private static final class constants (for naming descriptiveness) 
MAX_RATING = 5 and AVERAGE_RATING = 3 as what they represent ?